src/app/tests/TestWriteInteraction.cpp

```diff
@@ -61,7 +63,9 @@ class TestWriteInteraction
     static void TestWriteRoundtripWithClusterObjects(nlTestSuite * apSuite, void * apContext);
     static void TestWriteRoundtripWithClusterObjectsVersionMatch(nlTestSuite * apSuite, void * apContext);
     static void TestWriteRoundtripWithClusterObjectsVersionMismatch(nlTestSuite * apSuite, void * apContext);
-
+#if CONFIG_BUILD_FOR_HOST_UNIT_TEST
```

Comment:
This seems very fragile and easy to end up with us not running this when we should...

If we want to skip this via define, we should have some sort of "unit tests on nrfconnect" define and skip only if that is defined.

The idea of just making the set of data to send much bigger seemed like a better one to me.  What was wrong with it?

Reply:
Per Slack discussion: the WriteClient constructor actually behaves differently depending on `CONFIG_BUILD_FOR_HOST_UNIT_TEST`, so this test is completely broken if that's ever not defined....

We need to fix things so that's not the case, but that is not part of this PR.  Needs a followup.

Reply:
https://github.com/project-chip/connectedhomeip/issues/21737